docs/RFCs/19042021_self_signed_certificates.md

```diff
@@ -0,0 +1,240 @@
+ 
+## Goals
+ 
+* Helm install command should be self-sufficient to launch the CockroachDB cluster in a secure mode.
+ 
+* Dependency on external cert-manager should not be mandatory for creating CockroachDB cluster in a secure mode.
+ 
+* Manual steps should not be required for creating the CockroachDB cluster in a secure mode.
+  
+## Non-Goals
+ 
+* This RFC does not intend to fix the issues in the current default method of using Kubernetes CA
+ 
+## Helm Configuration
+This section specifies the suggested changes around user input in the Helm chart
+ 
+1. Add option specifying CockroachDB to manage the certificates, `tls.certs.generate.enabled` as true/false.  
+  Enabling this option will result in CockroachDB creating Node and Client certificates using a CA(either generated or provided by the user).
+ 
+2. Add option specifying CockroachDB to use user provided CA, `tls.certs.generate.caProvided` as true/false.  
+  Enabling this option will result in the generation of Node and Client certificates using the CA provided by the user.
+ 
+3. Add option specifying the secret name containing user provide CA, `tls.certs.generate.caSecret`.  
+  The secret name specified in this option will be used as a source for user-provided CA.
+  This option is mandatory if the `tls.certs.generate.caProvided` is true.
+ 
+4. Add option specifying the CA certificate expiration duration, `tls.certs.generate.caCertDuration`.  
+  This duration will only be used when we create our own CA. The duration value from this option will be used to set the expiry of the generated CA certificate.
+  By default, the CA expiry would be set to 10 years.
+ 
+5. Add option specifying the Client certificate expiration duration, `tls.certs.generate.clientCertDuration`.  
+  The duration value from this option will be used to set the expiry of the generated Client certificates. By default, Client certificate expiry would be set to 1 year.
+ 
+6. Add option specifying the Node certificate expiration duration, `tls.certs.generate.nodeCertDuration`.  
+  The duration value from this option will be used to set the expiry of the generated Node certificates. By default, Node certificate expiry would be set to 1 year.
+ 
+7. Add option specifying CockroachDB to manage rotation of the generated certificates, `tls.certs.generate.rotateCerts` as true/false.  
+  Enabling this option will result in auto-rotation of the certificates, before the expiry.
+ 
+## Helm Input Validation
+ 
+1. If `tls.certs.generate.caProvided` is set to true, then value for `tls.certs.generate.caSecret` must be provided.
+ 
+2. If value for `tls.certs.generate.caSecret` is provided, secret should exist in the CockroachDB install namespace.
+ 
+3. Value for `tls.certs.generate.caCertDuration` should be at least two months greater than the value for `tls.certs.generate.clientCertDuration`
```

Comment:
Have we finalised 2 month or we need to make cacert duration double of cert duration

Reply:
done